prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -109,3 +137,22 @@ class DeployableMicrochainModifiableSystemPromptAgent3(
 ):
     task_description = AgentIdentifier.MICROCHAIN_AGENT_OMEN_LEARNING_3
     model = SupportedModel.llama_31_instruct
+
+
+class DeployableMicrochainWithGoalManagerAgent0(DeployableMicrochainAgent):
+    task_description = AgentIdentifier.MICROCHAIN_AGENT_OMEN_WITH_GOAL_MANAGER
+    model = SupportedModel.gpt_4o
+    goal_manager = GoalManager(
```

Comment:
I think this would bring the same issue as I had with TavilyStorage. If GoalManager is initialized on import-level, then it will also be initialized in agent runs that don't use it and don't have Postgres connection string in the environment variables (CoinFlip agent, for example), and because of it, the run will fail.

I fixed it by initialising it in the `load` method https://github.com/gnosis/prediction-market-agent/pull/409/files:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/94d0907d-8c76-407d-b6ad-9e7e6657b471)

Reply:
Ah good point. I've moved this into a function that's called during `run` instead of on init, but inside `DeployableMicrochainAgent.build_goal_manager` instead (which is called after the agent is built, to also solve the problem of not hardcoding the goal manager's 'capabilities')